features/deep-linking/deep-link-handler.component.tsx

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useEffect } from "react"
+import { Linking } from "react-native"
+import { IXmtpInboxId } from "@/features/xmtp/xmtp.types"
+import { useAppState } from "@/stores/use-app-state-store"
+import { logger } from "@/utils/logger"
+import { parseURL } from "./link-parser"
+import { useConversationDeepLinkHandler } from "./conversation-navigator"
+
+/**
+ * Component that handles deep links for the app
+ * This should be included at the app root to handle incoming links
+ */
+export function DeepLinkHandler() {
+  const { currentState } = useAppState.getState()
+  const { handleConversationDeepLink } = useConversationDeepLinkHandler()
+
+  /**
+   * Handle a URL by parsing it and routing to the appropriate handler
+   */
+  const handleUrl = useCallback(async (url: string) => {
+    logger.info(`Handling deep link URL: ${url}`)
+    
+    const { segments, params } = parseURL(url)
+    
+    // Handle different types of deep links based on the URL pattern
+    if (segments[0] === "conversation" && segments[1]) {
+      // Pattern: converse://conversation/{inboxId}
+      const inboxId = segments[1] as IXmtpInboxId
+      const composerTextPrefill = params.composerTextPrefill
+      
+      logger.info(`Deep link matches conversation pattern, inboxId: ${inboxId}${
+        composerTextPrefill ? `, composerTextPrefill: ${composerTextPrefill}` : ''
+      }`)
+      
+      // Use the conversation deep link handler
+      await handleConversationDeepLink(inboxId, composerTextPrefill)
```

Comment:
let's follow the pattern of passing objects so like `handleConversationDeepLink({...})`